src/server/main_service.cc

```diff
@@ -1091,7 +1093,9 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
     if (cmd_name == "SELECT" || absl::EndsWith(cmd_name, "SUBSCRIBE"))
       return ErrorReply{absl::StrCat("Can not call ", cmd_name, " within a transaction")};
 
-    if (cmd_name == "WATCH" || cmd_name == "FLUSHALL" || cmd_name == "FLUSHDB")
+    // for some reason we get a trailing \n\r, and that's why we use StartsWith
```

Comment:
do we always have non empty `tail_args` here? 
can you dcheck?

Reply:
I don't think we need a `dcheck` because we call `Validate` on the command a few lines above and it internally checks for its arity. `CLIENT` requires at least one argument, so if the `cmd_name` is client we know it `tail_args[0]` is a valid index. 

If you till think I am wrong or I missed something I am happy to add it :)

Reply:
DCHECK is a replacement for a comment. What you wrote here is not immediately clear to me from reading the code. When you add a DCHECK, it is clear to a reader that somewhere else something handles the corner cases so that code here can not have empty tail_args. if someone else will change the code there and the invariant won't hold we might catch it with dcheck.